plugin/src/main/java/org/opensearch/ml/action/upload_chunk/MLModelChunkUploader.java

```diff
@@ -0,0 +1,159 @@
+                                        .modelConfig(existingModel.getModelConfig())
+                                        .totalChunks(existingModel.getTotalChunks())
+                                        .modelContentHash(existingModel.getModelContentHash())
+                                        .modelContentSizeInBytes(existingModel.getModelContentSizeInBytes())
+                                        .createdTime(existingModel.getCreatedTime())
+                                        .build();
+                                    IndexRequest indexReq = new IndexRequest(ML_MODEL_INDEX);
+                                    indexReq.id(modelId);
+                                    indexReq
+                                        .source(
+                                            mlModelMeta
+                                                .toXContent(XContentBuilder.builder(XContentType.JSON.xContent()), ToXContent.EMPTY_PARAMS)
+                                        );
+                                    indexReq.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+                                    client.index(indexReq, ActionListener.wrap(re -> {
+                                        log.debug("Index model successful", existingModel.getName());
+                                        semaphore.release();
+                                    }, e -> {
+                                        log.error("Failed to update model state", e);
+                                        semaphore.release();
+                                        listener.onFailure(e);
+                                    }));
+                                }
+                                listener.onResponse(new MLUploadModelChunkResponse("Uploaded"));
+                            }, e -> {
+                                log.error("Failed to upload chunk model", e);
+                                listener.onFailure(e);
+                            }));
+                        }, ex -> {
+                            log.error("Failed to init model index", ex);
+                            listener.onFailure(ex);
+                        }));
+                    } catch (Exception e) {
+                        log.error("Failed to parse ml model" + r.getId(), e);
+                        listener.onFailure(e);
+                    }
+                } else {
+                    listener.onFailure(new MLResourceNotFoundException("Failed to find model"));
+                }
+            }, e -> {
+                if (e instanceof IndexNotFoundException) {
+                    listener.onFailure(new MLResourceNotFoundException("Failed to find model"));
+                } else {
+                    log.error("Failed to get ML model " + modelId, e);
+                    listener.onFailure(e);
+                }
+            }));
+        } catch (Exception e) {
+            log.error(e.getMessage());
```

Comment:
This way won't print exception stack trace, suggest change to `log.error("Fail to upload chunk for model " + modelId, e);`